priam/src/main/java/com/netflix/priam/backupv2/ForgottenFilesManager.java

```diff
@@ -125,22 +131,74 @@ public void findAndMoveForgottenFiles(Instant snapshotInstant, File snapshotDir)
         return FileUtils.listFiles(columnfamilyDir, fileFilter, null);
     }
 
-    protected void moveForgottenFiles(File columnfamilyDir, Collection<File> columnfamilyFiles) {
+    protected void moveForgottenFiles(File columnfamilyDir, Collection<File> columnfamilyFiles)
+            throws IOException {
+        // This is a list of potential forgotten file(s). Note that C* might still be using
+        // files as part of read, so we really do not want to move them until we meet the
+        // @link{IConfiguration#getForgottenFileGracePeriodDaysForRead} window elapses.
+
         final Path destDir = Paths.get(columnfamilyDir.getAbsolutePath(), LOST_FOUND);
-        for (File file : columnfamilyFiles) {
-            logger.warn(
-                    "Forgotten file: {} found for CF: {}",
-                    file.getAbsolutePath(),
-                    columnfamilyDir.getName());
-            if (config.isForgottenFileMoveEnabled()) {
-                try {
-                    FileUtils.moveFileToDirectory(file, destDir.toFile(), true);
-                } catch (IOException e) {
-                    logger.error(
-                            "Exception occurred while trying to move forgottenFile: {}. Ignoring the error and continuing with remaining backup/forgotten files.",
-                            file);
-                    e.printStackTrace();
+        FileUtils.forceMkdir(destDir.toFile());
+        final Collection<Path> columnfamilyPaths =
+                columnfamilyFiles
+                        .parallelStream()
+                        .map(file -> Paths.get(file.getAbsolutePath()))
+                        .collect(Collectors.toList());
+
+        for (Path file : columnfamilyPaths) {
+            try {
+                final Path symbolic_link =
+                        Paths.get(destDir.toFile().getAbsolutePath(), file.toFile().getName());
+                // Lets see if there is a symbolic link to this file already?
+                if (!Files.exists(symbolic_link)) {
+                    // If not, lets create one and work on next file.
+                    Files.createSymbolicLink(symbolic_link, file);
+                    continue;
+                } else if (Files.isSymbolicLink(symbolic_link)) {
+                    // Symbolic link exists, is it older than our timeframe?
+                    Instant last_modified_time =
+                            Files.getLastModifiedTime(symbolic_link, LinkOption.NOFOLLOW_LINKS)
+                                    .toInstant();
+                    if (DateUtil.getInstant()
+                            .isAfter(
+                                    last_modified_time.plus(
+                                            config.getForgottenFileGracePeriodDaysForRead(),
+                                            ChronoUnit.DAYS))) {
+                        // Eligible for move.
+                        logger.warn(
```

Comment:
Do you want to move this warn line to below  `config.isForgottenFileMoveEnabled()`. Currently, there is no way for us to track which files actually got moved to LOST_FOUND directory.